geode-core/src/test/java/org/apache/geode/internal/tcp/ConnectionTransmissionTest.java

```diff
@@ -0,0 +1,258 @@
+import org.apache.geode.distributed.internal.membership.InternalDistributedMember;
+import org.apache.geode.distributed.internal.membership.api.Membership;
+import org.apache.geode.internal.monitoring.ThreadsMonitoring;
+import org.apache.geode.internal.monitoring.executor.AbstractExecutor;
+import org.apache.geode.internal.net.BufferPool;
+import org.apache.geode.internal.net.SocketCloser;
+import org.apache.geode.test.junit.categories.MembershipTest;
+
+@Category(MembershipTest.class)
+public class ConnectionTransmissionTest {
+
+  /**
+   * Create a sender connection and a receiver connection and pass data from
+   * one to the other.
+   *
+   * This test uses a real socket, but attempts to mock all other collaborators
+   * of connection, such as the InternalDistributedSystem.
+   */
+  @Test
+  public void testDataTransmittedBetweenSenderAndReceiverIfMembershipCheckPassed()
+      throws Exception {
+    final DMStats stats = mock(DMStats.class);
+    final BufferPool bufferPool = new BufferPool(stats);
+
+    final ServerSocketChannel acceptorSocket = createReceiverSocket();
+    final int serverSocketPort = acceptorSocket.socket().getLocalPort();
+
+    final CompletableFuture<Connection> readerFuture = createReaderFuture(acceptorSocket, true);
+
+    // Create a sender that connects to the server socket, which should trigger the
+    // reader to be created
+    final Connection sender = createWriter(serverSocketPort, false);
+    // Get the reader from the future
+    final Connection reader = readerFuture.get();
+
+    final ReplyMessage msg = createRelyMessage(sender);
+
+    final List<Connection> connections = new ArrayList<>();
+    connections.add(sender);
+
+    final BaseMsgStreamer streamer = MsgStreamer.create(connections, msg, false, stats, bufferPool);
+    streamer.writeMessage();
+
+    await().untilAsserted(() -> verify(reader, times(1)).readMessage(any(), any()));
+
+    assertThat(reader.isClosing()).isFalse();
+    verify(reader, times(1)).readHandshakeForReceiver(any());
+    verify(reader, times(0)).requestClose(any());
+  }
+
+  @Test
+  public void testReceiverClosesConnectionIfMembershipCheckFailed() throws Exception {
+    final DMStats stats = mock(DMStats.class);
+    final BufferPool bufferPool = new BufferPool(stats);
+    final ServerSocketChannel acceptorSocket = createReceiverSocket();
+
+    final int serverSocketPort = acceptorSocket.socket().getLocalPort();
+    final CompletableFuture<Connection> readerFuture = createReaderFuture(acceptorSocket, false);
+
+    final Connection sender = createWriter(serverSocketPort, true);
+
+    final Connection reader = readerFuture.get();
+    final ReplyMessage msg = createRelyMessage(sender);
+
+    final List<Connection> connections = new ArrayList<>();
+    connections.add(sender);
+
+    final BaseMsgStreamer streamer = MsgStreamer.create(connections, msg, false, stats, bufferPool);
+    streamer.writeMessage();
```

Comment:
All the setup in this method from about line 102 to about line 118 is identical to logic in the previous test (with the exception of the `isSenderInView` parameter to `createReaderFuture()`.

Would it make sense to turn that duplicated code into a method that takes a `isSenderInView` parameter?